1095 distinguish openid providers

cmeessen opened the pull request:
# Distinguish between login and coupling auth providers

Fixes #1095 #1097

Changes proposed in this pull request:

* Distinguish between auth providers used for login and for account coupling. This allows couple an ORCID account to an RSD account without having to allow ORCID login simultaneously
* :exclamation: This PR adds a new environment variable `RSD_AUTH_COUPLE_PROVIDERS`
* contains minor fix: removed the port mapping for the frontend dev mode because this was not necessary

How to test:

* in `.env`, set:
  ```
  RSD_AUTH_PROVIDERS="SURFCONEXT;LOCAL"
  RSD_AUTH_COUPLE_PROVIDERS=ORCID
  ```
* `docker compose build --parallel && docker compose up --scale scrapers=0`
* open login dialogue and verify that providers are Surf and Local only
* login
* navigate to user profile and link an ORCID account to the rsd account
* `docker compose down`
* add `ORCID` to `RSD_AUTH_PROVIDERS` and repeat previous steps. Verify that ORCID login works
* `docker compose down`
* in `.env`, comment out `RSD_AUTH_COUPLE_PROVIDERS`, or set it to something else than `ORCID`
* launch RSD and verify that ORCID login works but coupling button is not shown
* open the user documentation http://localhost/documentation/rsd-instance/configurations/ and check for updated content about the ORCID integration

PR Checklist:

*   [ ] Increase version numbers in `docker-compose.yml`
*   [x] Link to a GitHub issue
*   [x] Update documentation
*   [ ] Tests

cmeessen commented:
Waiting for #1103 to be closed by @ewan-escience

ewan-escience commented on authentication/src/main/java/nl/esciencecenter/rsd/authentication/Config.java:
```diff
@@ -22,7 +22,7 @@ public static String jwtSigningSecret() {
 	}
 
 	private static Collection<String> rsdAuthProviders() {
```
I would rename this method to `rsdAuthCoupleProviders`

cmeessen replied:
Good point, I updated this.

ewan-escience reviewed (changes requested):
Looks good, just one small renaming needed.

ewan-escience commented:
It is also good to note that when ORCID login is disabled and ORCID coupling is enabled, when an ORCID is coupled, this ORCID is still added to the login allow list.

cmeessen commented:
> It is also good to note that when ORCID login is disabled and ORCID coupling is enabled, when an ORCID is coupled, this ORCID is still added to the login allow list.

I added a warning about this into the documentation, and also split it up into sections for common variables, coupling and authentication respectively. Could you please double check whether all common variables are mentioned correctly?

ewan-escience commented:
> Could you please double check whether all common variables are mentioned correctly?

Everything seems to be there. You used as example`ORCID_REDIRECT_COUPLE=http://YOUR-RSD-DOMAIN/auth/couple/orcid`, maybe you can also use `YOUR-RSD-DOMAIN` for `ORCID_REDIRECT`.

dmijatovic commented on .env.example:
```diff
@@ -71,7 +71,13 @@ RSD_ENVIRONMENT=prod
 # if env value is not provided default provider is set to be SURFCONEXT
 # if you add the value "LOCAL", then local accounts are enabled, USE THIS FOR TESTING PURPOSES ONLY
-RSD_AUTH_PROVIDERS=SURFCONEXT;ORCID;AZURE;LOCAL
+RSD_AUTH_PROVIDERS="SURFCONEXT;ORCID;AZURE;LOCAL"
```
Do we need `"`? I think it works with or without `"` and we do not use in other variables.

cmeessen replied:
I think I added it because I had an issue with the `;` being interpreted as a command separator. But I couldn't reproduce it and removed the `"`.
